lstchain/scripts/lstchain_dl1ab.py

```diff
@@ -158,19 +152,29 @@ def main():
                     length = np.rad2deg(np.arctan2(dl1_container.length, foclen))
                     dl1_container.width = width
                     dl1_container.length = length
-                    dl1_container.r = np.sqrt(dl1_container.x ** 2 + dl1_container.y ** 2)
                     dl1_container.log_intensity = np.log10(dl1_container.intensity)
 
-                disp_dx, disp_dy, disp_norm, disp_angle, disp_sign = disp(dl1_container['x'].to_value(u.m),
-                                                                          dl1_container['y'].to_value(u.m),
-                                                                          params['src_x'][ii],
-                                                                          params['src_y'][ii],
-                                                                          )
-                dl1_container['disp_dx'] = disp_dx
-                dl1_container['disp_dy'] = disp_dy
-                dl1_container['disp_norm'] = disp_norm
-                dl1_container['disp_angle'] = disp_angle
-                dl1_container['disp_sign'] = disp_sign
+                if ['disp_dx', 'disp_dy', 'disp_norm', 'disp_angle', 'disp_sign'] in dl1_params_input:
```

Comment:
This is not working as expected, either I check that only one of those `disp_` parameters are present or do something like `if disp_params.issubset(dl1_params_input)` which slows a lot the runtime. Any suggestion?

Reply:
This is a check that has to run once per file. How can that be slowing it down by a lot?

```
if set(dl1_params.intersection({'disp_dx', 'disp_dy', 'disp_norm', 'disp_angle', 'disp_sign'}):
```

Reply:
Also:

```

In [5]: %timeit set(dl1_columns).intersection({'disp_dx', 'disp_dy', 'disp_norm', 'disp_angle', 'disp_sign'})
777 ns ± 2.24 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

how can this be a problem?

Reply:
Dependening on if you want `all` behaviour or `any` behaviour (i would go for `any`), use `intersection` or `issubset`

Reply:
I was doing that inside the loop over all the events `for ii, row in enumerate(image_table)`

Reply:
Even then. It takes less than *a microsecond*`. But why do you do that in the loop? The `parameters_to_update` will be fixed for the whole file, or not?

Reply:
Yes, you are right. I will refactor that.

I got the impression in my first tests that it was much slower but then tested with the another `if 'disp_dx'...` condition and it was pretty much the same.

Reply:
Done